lti_consumer/oauth.py

```diff
@@ -0,0 +1,153 @@
+
+
+def get_oauth_request_signature(key, secret, url, headers, body):
+    """
+    Returns Authorization header for a signed oauth request.
+
+    Arguments:
+        key (str): LTI provider key
+        secret (str): LTI provider secret
+        url (str): URL for the signed request
+        header (str): HTTP headers for the signed request
+        body (str): Body of the signed request
+
+    Returns:
+        str: Authorization header for the OAuth signed request
+    """
+    client = oauth1.Client(client_key=unicode(key), client_secret=unicode(secret))
+    try:
+        # Add Authorization header which looks like:
+        # Authorization: OAuth oauth_nonce="80966668944732164491378916897",
+        # oauth_timestamp="1378916897", oauth_version="1.0", oauth_signature_method="HMAC-SHA1",
+        # oauth_consumer_key="", oauth_signature="frVp4JuvT1mVXlxktiAUjQ7%2F1cw%3D"
+        __, headers, __ = client.sign(
+            unicode(url.strip()),
+            http_method=u'POST',
+            body=body,
+            headers=headers
+        )
+    except ValueError:  # Scheme not in url.
+        raise LtiError("Failed to sign oauth request")
+
+    return headers['Authorization']
+
+
+def verify_oauth_body_signature(request, lti_provider_secret, service_url):
+    """
+    Verify grade request from LTI provider using OAuth body signing.
+
+    Uses http://oauth.googlecode.com/svn/spec/ext/body_hash/1.0/oauth-bodyhash.html::
+
+        This specification extends the OAuth signature to include integrity checks on HTTP request bodies
+        with content types other than application/x-www-form-urlencoded.
+
+    Arguments:
+        request (xblock.django.request.DjangoWebobRequest): Request object for current HTTP request
+        lti_provider_secret (str): Secret key for the LTI provider
+        service_url (str): URL that the request was made to
+        content_type (str): HTTP content type of the request
+
+    Raises:
+        LtiError if request is incorrect.
+    """
+
+    headers = {
+        'Authorization': unicode(request.headers.get('Authorization')),
+        'Content-Type': request.content_type,
+    }
+
+    sha1 = hashlib.sha1()
+    sha1.update(request.body)
+    oauth_body_hash = base64.b64encode(sha1.digest())  # pylint: disable=E1121
+    oauth_params = oauth1.rfc5849.signature.collect_parameters(headers=headers, exclude_oauth_signature=False)
+    oauth_headers = dict(oauth_params)
+    oauth_signature = oauth_headers.pop('oauth_signature')
+    mock_request_lti_1 = Mock(
+        uri=unicode(urllib.unquote(service_url)),
+        http_method=unicode(request.method),
+        params=oauth_headers.items(),
+        signature=oauth_signature
+    )
+    mock_request_lti_2 = Mock(
+        uri=unicode(urllib.unquote(request.url)),
+        http_method=unicode(request.method),
+        params=oauth_headers.items(),
+        signature=oauth_signature
+    )
+    if oauth_body_hash != oauth_headers.get('oauth_body_hash'):
```

Comment:
I know this code is copied over from lti_module.py. But is there an easy way to accomplish the same without using `Mock` here? Coders generally don't expect to see it in non-test code. A nit.